autotest/test_canopy_runoff.py

```diff
@@ -39,19 +38,18 @@ def test_init(self, domain, control, params, tmp_path):
         ans = {}
         for key in comparison_var_names:
             nc_pth = output_dir / f"{key}.nc"
-            ans[key] = adapter_factory(nc_pth, variable_name=key)
+            ans[key] = adapter_factory(
+                nc_pth, variable_name=key, control=control
+            )
```

Comment:
adapter_factory takes now control which it passes to any adapter that tracks time in the advance (eg. AdapterNetcdf)